File: 18Jan2024.py
# Problem Link :- https://leetcode.com/problems/climbing-stairs/

class Solution:
    def climbStairs(self, n: int) -> int:
        list = [0]*(n+5)
        list[n]=1
        for i in range(n-1,-1,-1) :
            list[i]=list[i+1]+list[i+2]
        return list[0]

File: 18Jan2024.java
// Problem Link :- https://leetcode.com/problems/climbing-stairs/

class Solution {
    public int climbStairs(int n) {
        int[] ans = new int[n + 5];
        ans[n] = 1;
        for (int i = n - 1; i >= 0; i--) {
            ans[i] = ans[i + 1] + ans[i + 2];
        }
        return ans[0];
    }
}